Userland/Applications/PixelPaint/Tools/PolygonalSelectTool.cpp

```diff
@@ -33,6 +33,9 @@ void PolygonalSelectTool::flood_polygon_selection(Gfx::Bitmap& polygon_bitmap, G
     polygon_bitmap.flood_visit_from_point({ polygon_bitmap.width() - 1, polygon_bitmap.height() - 1 }, 0, move(pixel_reached));
 
     selection_mask.shrink_to_fit();
+    if (selection_mask.bounding_rect().width() == 1 || selection_mask.bounding_rect().height() == 1)
+        selection_mask = {};
```

Comment:
Instead of repeating this logic for all possible (future) selection tools, you could create a reused method somewhere to shrink the mask down to `0x0` and test for `.is_empty()` before applying the selection here.